### PR TITLE
Earth 1021 more social

### PR DESCRIPTION
**Pull request by buttonwillowsix**
# READY

# Summary
- Adding social media from Get Social to events and news. 
- NOT Spotlights yet since that php error isn't resolved.
- Wrap location text in a link using the MapURL

# Needed By (Date)
- Next push, or could be at same time as Spotlights (EARTH-1080)

# Urgency
- Not very, but would be nice.

# Steps to Test

1. Go to Manage Display for Events and drag the MapURL field into the new MapURL section.  Format it as a Link, make sure the Trim Length field is blank and you have checked "URL Only" and "Show URL as Plain Text."
2. Go to an event node. See that the teal social media icons have been replaced by the getsocial counts.
2.5. Check that the there is a URL in the MapURL field. If there is, check that the "Where" link goes there.
3. Go to a news node. See that the teal social media icons have been replaced by the getsocial counts.
4. Test both at various screensizes

# Affected Projects or Products
- News, Events

# Associated Issues and/or People
- EARTH-1021
- EARTH-1080
- EARTH-1064

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: templates/stanford_news/field--node--field-s-news-date.html.twig [hunk 1]
```diff
@@ -41,38 +41,7 @@
...
-    </div>
-    <div class="field--field-s-news-date__action email-popup">
-      <a href="mailto:?subject={{ 'I wanted you to see this news article'|t }}&amp;body={{ 'Read more about:'|t }} at {{ node_title|render|striptags|trim }} {{ url("<current>") }}" title="{{ 'Share by Email'|t }}">
-        <svg class="i-svg" role="img">
-          <title>{{ 'mail'|t }}</title>
-          <use xlink:href="#i-mail" xmlns:xlink="http://www.w3.org/1999/xlink"></use>
-        </svg>
-      </a>
-    </div>
+    <div class="getsocial gs-inline-group"></div>
```
Comment (sherakama) [inline templates/stanford_news/field--node--field-s-news-date.html.twig:44]: Wow, this slimmed down.

File: templates/stanford_event/stanford-event.html.twig [hunk 2]
```diff
@@ -59,7 +24,13 @@
         {% if where|render|striptags|trim is not empty %}
         <dt>{{ 'Where'|t }}:</dt>
         <dd>
-          {{ where }}
+          {% if mapurl|render|striptags|trim is not empty %}
+          <A href="{{ mapurl }}">
```
Comment (sherakama) [inline templates/stanford_event/stanford-event.html.twig]: lowercase tags please.

File: templates/html.html.twig [hunk 3]
```diff
@@ -47,6 +47,13 @@
     <title>{{ head_title|safe_join(' | ') }}</title>
     <css-placeholder token="{{ placeholder_token }}">
     <js-placeholder token="{{ placeholder_token }}">
+    <script>
```
Comment (sherakama) [inline templates/html.html.twig]: This should go into a script file and added through a library. I can show you this if you want to zoom session this change.

File: templates/stanford_event/stanford-event.html.twig (hunk 2 above)
Comment (sherakama) [inline templates/stanford_event/stanford-event.html.twig]: May was well output `{{ mapurl|render|striptags|trim }}` inside the href property.

File: scss/theme/stanford-news.scss [hunk 4]
```diff
@@ -354,21 +369,26 @@ figure.align-center .figure-container {
...
   @include grid-media($media-lg) {
     @include grid-column(3 of 12);
-    margin-left: 20px;
+    margin-left: 30px;
   }
 
   a {
     display: block;
-    width: 100%;
+    //width: 100%;
```
Comment (sherakama) [inline scss/theme/stanford-news.scss]: Remove commented out code.

**Review by sherakama — CHANGES_REQUESTED** [review]
Let me know if I can help you with the javascript library.
https://www.drupal.org/docs/8/theming/adding-stylesheets-css-and-javascript-js-to-a-drupal-8-theme

File: templates/stanford_news/field--node--field-s-news-date.html.twig (hunk 1 above)
Comment (buttonwillowsix) [reply templates/stanford_news/field--node--field-s-news-date.html.twig:44]: Yeah, that is my favorite part.

File: templates/html.html.twig (hunk 3 above)
Comment (buttonwillowsix) [reply templates/html.html.twig]: Oops, I can actually take this out.  I am not sure that I really need to have it on the live site at all? I think Google Tag Manager might be handling it. Hm, although maybe better safe than sorry...

Comment (buttonwillowsix) [reply templates/html.html.twig]: Will take you up on your offer. Assuming it goes in js/vendor ?

File: js/source/getsocial.js [hunk 5]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (sherakama) [inline js/source/getsocial.js]: Empty line at beginning of file.

File: js/source/getsocial.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+(function() {
+var po = document.createElement('script'); po.type = 'text/javascript'; po.async = true;
```
Comment (sherakama) [inline js/source/getsocial.js]: function body should be indented 2 spaces.

File: matson.theme [hunk 7]
```diff
@@ -133,7 +133,7 @@ function matson_page_attachments_alter(array &$attachments) {
 
       case 'stanford_news':
         $attachments['#attached']['library'][] = 'matson/stanford_news';
-        $attachments['#attached']['library'][] = 'matson/sharrre';
+        $attachments['#attached']['library'][] = 'matson/getsocial';
```
Comment (sherakama) [inline matson.theme:136]: Nice.

File: templates/stanford_event/stanford-event.html.twig [hunk 8]
```diff
@@ -2,42 +2,7 @@
...
-          <a href="mailto:?subject={{ 'I wanted you to see this news article'|t }}&amp;body={{ 'Read more about:'|t }} at {{ node_title|render|striptags|trim }} {{ url("<current>") }}" title="{{ 'Share by Email'|t }}">
-            <svg class="i-svg" role="img">
-              <title>{{ 'mail'|t }}</title>
-              <use xlink:href="#i-mail" xmlns:xlink="http://www.w3.org/1999/xlink"></use>
-            </svg>
-          </a>
-        </div>
-      </div>
-    </div>
+
```
Comment (sherakama) [inline templates/stanford_event/stanford-event.html.twig:5]: Not adding: `<div class="getsocial gs-inline-group"></div>` to this?

Comment (buttonwillowsix) [reply templates/stanford_event/stanford-event.html.twig:5]: Oops, I will!
However, attaching the library isn't yet working, so need to fix that too.

Comment (buttonwillowsix) [reply templates/stanford_event/stanford-event.html.twig:5]: Oh, also we added it further down. It looks weird in the spot we had it at smaller screen sizes.

**Comment by buttonwillowsix** [comment]
@sherakama , something I have done isn't working. I must have missed a step in attaching the getsocial script?

**Comment by ksharp-drupal** [comment]
Saw the following behavior when testing:
1. checkout the branch and clear-cache
2. go to an events page in Firefox
image
3. Note "0 shares" and buttons on two rows.
4. Click a button (I tested with email and Twitter)
5. Dismiss popup or return to browser
pie2
6. "0 shares" is gone and buttons are all on one row.

**Comment by buttonwillowsix** [comment]
@ksharp-drupal, I wonder if it is getting a generic version of the script? I am am just happy you saw ANY buttons. I wasn't getting that on mine. I am going to go check to make sure our getsocial settings didn't set us back to the default look.

**Comment by sherakama** [comment]
This is working for me. Some style issues around the event page that should be addressed but overall really good. 

I would ask for one more thing to go along with this PR, the configuration for the mapurl field should be bundled into the event module and site config. 

Screen Shot 2019-03-13 at 11 46 42 PM
Screen Shot 2019-03-13 at 11 46 47 PM

**Comment by buttonwillowsix** [comment]
@sherakama and @ksharp-drupal, I adjusted the styling so that the wrapping isn't happening, so I think that side of things is all ready. 
@ksharp-drupal, can I pass this to you or work with you to do what Shea is asking for the MapURL?

**Comment by ksharp-drupal** [comment]
Sure, no problem.

**Review by ksharp-drupal — APPROVED** [review]

**Pull request merged**